File: C++/Leetcode/Array/buySellStockI.cpp
#include <iostream>
#include <vector>
#include <limits>

class Solution {
    public:
        int maxProfit(std::vector<int>& prices){
            int minimum = std::numeric_limits<int>::max();
            int profit = 0;

            for (int i = 0; i < prices.size(); ++i){
                int cProf = prices[i] - minimum;
                if (prices[i] < minimum){
                    minimum = prices[i];
                }
                if (cProf > profit){
                    profit = cProf;
                }
            }
            return profit;
        }
};

int main(){
    Solution solution_instance;
    std::vector<int> nums = {7,1,5,3,6,4};
    std::cout << solution_instance.maxProfit(nums) << std::endl;
    return 0;
}

File: Python/Leetcode/Array/buySellStockI.py
class Solution(object):
    def maxProfit(self, prices):
        maxPrice = 0
        minimum = prices[0]
        for i in range(1, len(prices)):
            if prices[i]-minimum > maxPrice:
                maxPrice = max(maxPrice, prices[i]-minimum)
            minimum = min(minimum, prices[i])
        return maxPrice
        
if __name__ == '__main__':
    solution_instance = Solution()
    solution_instance.maxProfit([7,6,4,3,1])